File: src/types/hardware.ts
export interface RuntimeProfile {
  cpuName: string;
  cpuLogicalCores: number;
  totalMemoryMb: number;
  hasAvx2: boolean;
  hasAvx512: boolean;
  hasGpu: boolean;
  gpuNames: string[];
  supportedAcceleration: string; // "CPU_AVX2" or "CPU_BASIC" for the V1 CPU bundle
  runtimeAvailable: boolean;
  runtimeVersion: string | null;
  runtimeBackends: string[];
  recommendedModelIds: string[];
  fallbackReason: string | null;
}

export type PresetType = "fast" | "balanced" | "accurate" | "custom";

export interface RuntimePresetPreference {
  schemaVersion: number;
  preset: PresetType;
  userOverrideModel: string | null;
}

export interface PresetResolution {
  preset: PresetType;
  targetModelId: string;
  targetBackend: string;
  isModelInstalled: boolean;
  fallbackReason: string | null;
  tradeoffDescription: string;
}

File: src-tauri/src/models/hardware.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct RuntimeProfile {
    pub cpu_name: String,
    pub cpu_logical_cores: u32,
    pub total_memory_mb: u64,
    pub has_avx2: bool,
    pub has_avx512: bool,
    pub has_gpu: bool,
    pub gpu_names: Vec<String>,
    pub supported_acceleration: String, // "CPU_AVX2" or "CPU_BASIC" for the V1 CPU bundle
    pub runtime_available: bool,
    pub runtime_version: Option<String>,
    pub runtime_backends: Vec<String>,
    pub recommended_model_ids: Vec<String>,
    pub fallback_reason: Option<String>,
}
